runtime/sema/type.go

```diff
@@ -2503,6 +2505,7 @@ func ArrayFilterFunctionType(memoryGauge common.MemoryGauge, elementType Type) *
 			},
 		},
 		ReturnTypeAnnotation: NewTypeAnnotation(NewVariableSizedType(memoryGauge, elementType)),
+		Purity:               FunctionPurityView,
```

Comment:
This also force the predicate to be view. Not sure if we want to require it, given it isn't necessary? Not a big deal, just wanted to flag it. In the future we could maybe have two variants of the function (until we have a means of abstracting over this).

Reply:
Yeah, as mentioned in the PR summary, I think this is the right call in this case because predicates are generally thought of as non-mutating operations. But I don't feel too strongly about it. We could relax it later if people complain.

Reply:
Sorry, I missed that! Sounds good 👍